server.py

```diff
@@ -1,90 +1,385 @@
+    # Stanford CoreNLP
+    # =====================================================================================
+
+    nlp = StanfordCoreNLP('http://localhost:9000')
+
+    properties = {'annotators': 'tokenize,ssplit',
+                  'outputFormat': 'json'}
+
+    if language == 'cmn':
+        properties['customAnnotatorClass.tokenize'] = 'edu.stanford.nlp.pipeline.ChineseSegmenterAnnotator'
+        properties['tokenize.model'] = 'edu/stanford/nlp/models/segmenter/chinese/ctb.gz'
+        properties['tokenize.sighanCorporaDict'] = 'edu/stanford/nlp/models/segmenter/chinese'
+        properties['tokenize.serDictionary'] = 'edu/stanford/nlp/models/segmenter/chinese/dict-chris6.ser.gz'
+        properties['tokenize.sighanPostProcessing'] = 'true'
+        properties['ssplit.boundaryTokenRegex'] = urllib.quote_plus('[!?]+|[。]|[！？]+')
+    elif language == 'spa':
+        properties['tokenize.language'] = 'es'
 
-    text = request.form['text']
-    logger.info( 'text-received: %s' % text )
+    output = nlp.annotate(text, properties=properties)
 
-    text = text.encode('ascii', 'ignore').strip()
-    text = sent_tokenize( text )
-    text = [ word_tokenize(t.strip()) for t in text ]
-    logger.info( 'text after ssplit & tokenize: %s' % str(text) )
+    pprint.pprint(output)
 
+    text_array = []
+    text_to_offset = {}
+    sentences = output['sentences']
+    for sent in sentences:
+        new = []
+        tokens = sent['tokens']
+        for word in tokens:
+            new.append(word['originalText'])
+        if word['originalText'] not in text_to_offset:
+            text_to_offset
+        text_array.append(new)
+
+    # =====================================================================================
+    text = text_array
+    logger.info('text after split & tokenize: %s' % str(text))
+
+    # DEMO MODE
     if mode == 'demo':
-        result = doc2json( annotator.annotate( text ) )
-        logger.info( result )
+        # retrieve the MIDs from the csv file
+        start_time1 = time.time()
+        inference, score = annotator.annotate(text, isDevMode=True)
+
+        logger.info(inference)
+
+        # Replace the offsets from the annotator with the offsets from Stanford
+
+        print("ANNOTATOR.ANNOTATE TAKES %s SECONDS" % (time.time() - start_time1))
+
+        # Linking part: Uncomment to use 
+        # =============
+        # if language == "eng":
+        #     start_time2 = time.time()
+        #     write_sentence_to_file(inference)
+        #     print("WRITE_SENTENCE_TO_FILE TAKES %s SECONDS" % (time.time() - start_time2))
+
+        start_time3 = time.time()
+        if len(score) > 1:
+            result = inference_to_json(inference, score[1])
+        else:
+            result = inference_to_json(inference, score[0])
+
+        result['notes'] = notes
+        print("DOC2JSONDEMO TAKES %s SECONDS" % (time.time() - start_time3))
+
+        # Linking part: Uncomment to use 
```

Comment:
Wrap this in abstract APIs, be it methods or classes, and put it in another PR for archive purpose. 
This approach writes to disk many times and loads the same content again. **I won't merge this to the master branch in any case.** I don't want my future co-workers read this and underrate my skills.
Moreover, the "run.sh" script is really ugly written.